### PR TITLE
Simnode refactoring

### PR DESCRIPTION
**Pull request by claudious96**
This PR refactors the sim node related files in order to prepare to use it in the future for testing.

The simnode now doesn't need a list of available commands hardcoded, as long as the files in nossim follow the guidelines:
command: `show mac-address table` -> filename: `show_mac_address_table.<format>`
In case a `*` is used it is replaced with `all` in the filename.
At the moment there is one exception that cannot be automatically retrieved and there will be more in the future but the majority of the filename can be deduced from the command.

The simnode will be run as:
`tests/utilities/simnode_ssh.py --nos panos [--nos-version 8.0.0 --hostname leaf01 --listening-port 10000]`
Where options in square brackets are optional. 

In addition, it brings a `simnode_rest ` that provides XML output only for panos. It can be extended to provide also JSON REST APIs.

It also introduces `symnode.py` that, given a yaml file that describes the devices, starts the related processes to simulate the specified devices.

### CODE REVIEW

**Review by ddutt — APPROVED** [review]

**Review by ddutt — APPROVED** [review]
Please verify there are no linting errors

**Review by ddutt — APPROVED** [review]

**Review by ddutt — CHANGES_REQUESTED** [review]
Can you split the addition to pyproject.toml and the corresponding poetry.lock and build/requirements.txt from this commit?

**Review by LucaNicosia — APPROVED** [review]

**Review by claudiolor — APPROVED** [review]

File: tests/utilities/simnode.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                "required": [
+                    "nos", "version", "hostname"
+                ]
+            }
+        }
+    }
+}
+
+
+def run_simnodes(devices: list):
```
Comment (ddutt) [inline tests/utilities/simnode.py:40]: Sorry for being pedantic, but can you please add docstrings?

Comment (claudious96) [reply tests/utilities/simnode.py:40]: No problems, should be done now

**Pull request merged**